pkg/util/abiutil/abiutil.go

```diff
@@ -34,12 +41,7 @@ func ConvertType(in interface{}, proto interface{}) (out interface{}, err error)
 	return
 }
 
-func UnpackBigInt(d abi.ABI, result []byte, name string) (*big.Int, error) {
-	values, err := d.Unpack(name, result)
-	if err != nil {
-		return nil, fmt.Errorf("failed to unpack %v name: %v", name, err)
-	}
-
+func UnpackBigInt(values []interface{}) (*big.Int, error) {
```

Comment:
Naming convention for `UnpackSomething` had more sense while these functions were unpacking value from abi. Could functions with new approach have different naming; perhaps `ConvertSomething`?